### ISSUE TITLE
Android 14 (Samsung A14) : service stop after backgrounded it

### ISSUE THREAD

**Issue by danchiche** [Rapsssito/react-native-background-actions#225]
Hi,
For testing I start a counter when I click to a button on the app.
When I close the app (put it in background), the counter freeze.
When I switch off the screen, and switch back on, the counter continue when expected.

This problem is constated on Samsung A14, but I think is due to Android 14+.
I have a samsung s9 on android 10 and no problem, when I close the app, counter continue normally.

I tested this on expo 49 with react-native-background-actions 3.0.1 and with expo 51 with react-native-background-actions 4.0.0, and both the same experience.

Do anyone know what can I do ?

Thanks

**Comment by danchiche** [comment]
Yes but like I said before, it's not a problem of react-native-background-actions 4.0.0
I tested with expo 49 with react-native-background-actions 3.0.1, and the same problem

**Comment by j-jonathan** [comment]
Are notifications allowed in the Android app settings?

**Comment by danchiche** [comment]
It's impossible to enable notification in my app on the samsung A14, other apps have the ability to enable it.
I try to find on the internet how to enable notifications on my app, but I don't have the restricted setting option.
On my samsung s9 notification is allowed, and I see the notification when there is a task in background.

**Comment by danchiche** [comment]
New update, I ask for notification permission, and now I am able to allow the permission on the A14.
I make a new test and the same bug, the counter stop 5 seconde after I close the app, and continue when I open it again (or I switch off and on the screen)

**Comment by jacopo-ferroni** [comment]
help, same issue!

**Comment by vzkharov** [comment]
The same, after app is backgrounded get the error `app is not responding`.
Faced with this issue only on Android 14

**Comment by vzkharov** [comment]
There are some observation:
- `foregroundServiceType=shortService` working on Android 14 only 15 minutes according to `developer.android.com`
- When I tried change to `location` it works fine on Android 14, but on Android 10 this service has unexpected behavior (e.g. it can work as 15 minutes or 8 hours) *P.S. Maybe it just problem with Xiaomi*

**Comment by vzkharov** [comment]
Maybe there are some assumptions how to manage stable work of this background-task?

**Comment by Dwikavindra** [comment]
> Maybe there are some assumptions how to manage stable work of this background-task?

If you look at the official docs
[fgservicestypes](https://developer.android.com/develop/background-work/services/fg-service-types)

shortService only runs for 3 minutes. I think it has something to do with that it's not supposed to run longer than that thus causing errors.

**Comment by tmaly1980** [comment]
@Dwikavindra I believe you are onto something!

**Comment by wojciechbilka** [comment]
I tired to run in on Android 14 with `specialUse`. It has the same issue as for other foreground service types. In my case it seems to work well if app is closed or in a foreground, but if I leave it in the background or lock the screen the task pauses in less then 10 seconds, if I go back to the app it resumes. If anyone has a solution for that please share it here.

**Comment by JeronimoGarciaBielsa** [comment]
Same problem

**Comment by pankuweb** [comment]
In androidmenifest.xml
Use 
    <uses-permission android:name="android.permission.FOREGROUND_SERVICE_REMOTE_MESSAGING" />
<service android:name="com.asterinet.react.bgactions.RNBackgroundActionsTask" android:foregroundServiceType="remoteMessaging" android:exported="false"/>

**Comment by Rapsssito** [comment]
Duplicate, see https://github.com/Rapsssito/react-native-background-actions/issues/176#issuecomment-1559113415
